src/Data/Foldable.js

```diff
@@ -0,0 +1,26 @@
+/* global exports */
+"use strict";
+
+// module Data.Foldable
+
+exports.foldrArray = function (f) {
+  return function (init) {
+    return function (xs) {
+      /* jshint maxparams: 2 */
+      return xs.reduceRight(function (acc, x) {
```

Comment:
I find JSPerf results a bit unnerving because tests on there usually use just one input; this was one of the motivations for [purescript-benchotron](https://github.com/hdgarrood/purescript-benchotron) (which is close to an initial release now).

In this case, it seems not to matter, though. I tested this in the same way, `foldr (+) 0`, with benchotron on Ubuntu / node v0.10.25, and the `reduceRight` version is still slower, but only by a factor of about 2.5:

![foldbench](https://cloud.githubusercontent.com/assets/1270186/7670139/b9c14a8a-fc90-11e4-8232-b81517fb0742.png)